File: fibo.f90
module fibonacci
contains
  recursive function fibo (n) result (f)
    integer, intent (in) :: n
    integer :: f
    if (n < 2) then
       f = n
    else
       f = fibo (n-1) + fibo (n-2)
    end if
  end function fibo
end module fibonacci

program fibTest
  use fibonacci

  integer :: i
  character (len = 32) :: arg

  call getarg (1, arg)
  read (arg, '(i10)') i

  print *, fibo (i)
end program fibTest

File: fibo.pas
program fibonacci;
var
   num: integer;

function fibo(n : integer) : longint;
begin
   if n < 2 then fibo := n
   else fibo := fibo (n-2) + fibo (n-1);
end;

begin
   if ParamCount = 0 then
   begin
      num := 1;
   end
   else
   begin
      val (ParamStr (1), num);
   end;
   begin
      writeln (fibo(num));
   end

end.
